src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,90 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User user = new User("validLogin", "password123", 20);
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = new User("existingUser", "password123", 20);
+        Storage.people.add(user);
+        User newUser = new User("existingUser", "password123", 20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(newUser);
+        });
+    }
+
+    @Test
+    public void register_loginTooShort_notOk() {
+        User user = new User("short", "password123", 20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void register_passwordTooShort_notOk() {
+        User user = new User("validLogin", "pass", 20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void register_ageLessThan18_notOk() {
+        User user = new User("validLogin", "password123", 17);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
```

Comment:
This test checks that a `RegistrationException` is thrown when trying to register a user who is under 18. However, the test does not specify why 18 is the minimum age. It would be clearer if you defined a constant for the minimum age and used that in your test.